crates/iceberg/src/spec/values.rs

```diff
@@ -554,13 +554,14 @@ impl From<&Literal> for JsonValue {
                 PrimitiveLiteral::Decimal(_) => todo!(),
             },
             Literal::Struct(s) => {
```

Comment:
Hey @liurenjie1024 how should we design this method ? Since the iterator only return the value instead of the `(id, value, name)` tuple. I'm having trouble setting up the test for this method.

Reply:
Sorry for late reply. Is this method really useful? I think we should expose [this method](https://github.com/apache/iceberg-rust/blob/28d7006ba3af12b1f99dc078780b08ccdd8e93b1/crates/iceberg/src/spec/values.rs#L746)

Reply:
Isn't the this `Struct` method supposed to return a hashmap ? The StructType for iceberg requires a NestedField with the id and value.

Reply:
I think it's supposed to return an json object, and the field names should match struct field names. But this is not feasible without schema. [This method](https://github.com/apache/iceberg-rust/blob/28d7006ba3af12b1f99dc078780b08ccdd8e93b1/crates/iceberg/src/spec/values.rs#L746) is the correct approach I think.

Reply:
The Struct only has the `field` vector, I removed `field_ids` and `field_names`. I looked at `try_from_json` but I still couldn't figure out how to infer the schema from the Struct.

Reply:
> The Struct only has the field vector, I removed field_ids and field_names. I looked at try_from_json but I still couldn't figure out how to infer the schema from the Struct.

Sorry for the confusion, I mean this method should  be removed.

Reply:
I added a `todo!()` to the method instead, I tried to remove it completely but the compiler says that we need a match arm for the case where `value` is a Struct. Is this sufficient ? @liurenjie1024

Reply:
Sorry, I mean we should remove the total implementation of `From<&Literal> for JsonValue`. Since we have decided to remove type info for `Struct`, this method signature is incorrect. The conversion from/to json should delegate to ser/de module.

Reply:
`serde_json` doesn't seem to have a method to extract the JsonValue from the Literal. This was only possible through `From<&Literal> for JsonValue`

This line raises a compiler error when I removed `From<&Literal> for JsonValue`
```rust
fn check_json_serde(json: &str, expected_literal: Literal, expected_type: &Type) {
    ...
    let expected_json_value: JsonValue = (&expected_literal).into(); // error raised here
```

Reply:
Yes, after removing this, directly ser/de from json no longer works. The correct way to do this is to ser/de is using the ser/de module, you can see [this method](https://github.com/apache/iceberg-rust/blob/28d7006ba3af12b1f99dc078780b08ccdd8e93b1/crates/iceberg/src/spec/values.rs#L1654) as example. It delegates ser/de to serializaiton/deserialization system.

Reply:
So you are saying I should refactor `check_json_serde` to follow the structure of `check_convert_with_avro` ? 

Also just a reminder, but the code doesn't build when I remove `From<&values::Literal>` so I don't think removing it entirely will work.